### PR TITLE
Adding optional redis check to stockpile-wrapper

### PR DESCRIPTION
**Pull request by dry923**
The stockpile-wrapper will now optionally take a redis host and port. If they are not given nothing changes. If they are given and there is a valid uuid and node then it will attempt to check if metadata has already been gathered from the node. If it has then it just exits. If it has not, then it marks the node so it is not run again and then begins collecting data.

### CODE REVIEW

**Comment by bengland2** [comment]
Is there something that would give back a link to where the data is already stored in ES?    Don't need to copy the data, just get a pointer to it?   BUT if that other data set is removed, then the performance data in this dataset  has a dangling link to backpack data that no longer exists?

**Comment by dry923** [comment]
> Is there something that would give back a link to where the data is already stored in ES? Don't need to copy the data, just get a pointer to it? BUT if that other data set is removed, then the performance data in this dataset has a dangling link to backpack data that no longer exists?

@bengland2 I added a check to see if some basic data is there for the uuid and node. If it is actively running the collection, however, I let it be the only one running.

**Comment by rsevilla87** [comment]
@dry923 Maybe we can add a flag to "force-metadata-collection". I know we can force metadata collection if we don't pass redis arguments to stockpile, to be honest I'm not sure if it makes sense.
What do you think?

**Comment by dry923** [comment]
> @dry923 Maybe we can add a flag to "force-metadata-collection". I know we can force metadata collection if we don't pass redis arguments to stockpile, to be honest I'm not sure if it makes sense.
> What do you think?

@rsevilla87 Yeah I think its valid to be able to force it. The optional redis arguments are to allow it to still work outside of ripsaw with no change. 

I'll add something in shortly for the force flag.

**Review by rsevilla87 — APPROVED** [review]
LGTM, I like this feature, it will save us some time.

File: stockpile-wrapper/stockpile-wrapper.py [hunk 1]
```diff
@@ -73,6 +75,41 @@ def _run_stockpile():
...
+    # We are using metadata-cpuinfo as it is a basic index that should regularly be there without any extended permissions
+#    check_results = es.search(index="cpuinfo-metadata", body={'query': {'term': {'uuid.keyword': my_uuid, 'node_name.keyword': my_node}}}, size=1)
+    if check_results['hits']['total']['value'] > 0:
+        return True
+    else:
+        return False
+
+def _mark_node(r,my_node,my_uuid,server,port,check_val):
+    current_val = r.get(check_val)
+    print(current_val)
```
Comment (jtaleric) [inline stockpile-wrapper/stockpile-wrapper.py]: Did you mean to keep this? Possibly add more output if you want to keep it, so the reader of the log can figure out what this means..

Comment (dry923) [reply stockpile-wrapper/stockpile-wrapper.py]: I added that for debugging and it will be removed.

**Review by aakarshg — APPROVED** [review]
LGTM

**Review by jtaleric — APPROVED** [review]
LGTM

**Pull request merged**